File: In Java/src/A_Plus_B_Hard.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class A_Plus_B_Hard {
//HALF COMPLETE
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		
		int problems = Integer.parseInt(in.readLine());
		
		for (int i = 0; i < problems; i ++) {
			String answer = "";
			String[] temp = in.readLine().split(" ");
			String a = temp[0];
			String b = temp[1];
			String state = "";
			int carryOver = 0;
			
			if (a.charAt(0) != '-' && b.charAt(0) != '-') {
				state = "positive";
			} else if (a.charAt(0) == '-' && b.charAt(0) == '-') {
				state = "negative";
				a = a.substring(1);
				b = b.substring(1);
			} else if (a.charAt(0) == '-') {
				state = "subtract";
				a = a.substring(1);
			} else if (b.charAt(0) == '-') {
				state = "subtract";
				String temp1 = b;
				b = a;
				a = temp1;
				a = a.substring(1);
			}
			
			if (state.equals("positive") || state.equals("negative")) {
				while (a.length() > 0 || b.length() > 0) {
					if (a.length() > 0 && b.length() > 0) {
						int total = Integer.parseInt(a.substring(a.length()-1, a.length())) + Integer.parseInt(b.substring(b.length()-1, b.length())) + carryOver;
						answer = String.valueOf(total % 10) + answer;
						if (total >= 10)
							carryOver = (total - (total % 10)) / 10;
						else
							carryOver = 0;
						a = a.substring(0, a.length()-1);
						b = b.substring(0, b.length()-1);
					} else {
						if (b.length() > 0) {
							a = b;
							b = "";
						}
						if (a.length() > 0) {
							int total = Integer.parseInt(a.substring(a.length()-1, a.length())) + carryOver;
							answer = String.valueOf(total % 10) + answer;
							if (total >= 10)
								carryOver = (total - (total % 10)) / 10;
							else
								carryOver = 0;
							a = a.substring(0, a.length()-1);
						}
					}
				}
				
				if (state.equals("positive")) {
					if (carryOver != 0)
						System.out.println((String.valueOf(carryOver) + answer));
					else
						System.out.println(answer);
				} else {
					if (carryOver != 0)
						System.out.println("-" + (String.valueOf(carryOver) + answer));
					else
						System.out.println("-" + answer);
				}
			} else {
				//Subtraction. b - a
				BigInteger x = new BigInteger(b);
				BigInteger y = new BigInteger(a);
				System.out.println(x.subtract(y));
			}
		}

	}

}

File: In Python/A_plus_B_HARD.py
problems = int(input())
#NOT WORKING. USE JAVA SOLUTION!
for i in range(0, problems):
    answer = ''
    temp = input().split()
    num1 = temp[0]
    num2 = temp[1]
    carry_over = 0
    
    if (num1[0] != '-' and num2[0] != '-'):
        while (len(num1) > 0 or len(num2) > 0):
            if (len(num1) > 0 and len(num2) > 0):
                total = int(num1[-1]) + int(num2[-1]) + carry_over
                answer = str(total)[-1] + answer
                if total >= 10:
                    carry_over = int(str(total)[-2])
                num1 = num1[0:len(num1)-1]
                num2 = num2[0:len(num2)-1]
            else:
                if (len(num1) > 0):
                    total = int(num1[-1]) + carry_over
                    answer = str(total)[-1] + answer
                    if total >= 10:
                        carry_over = int(str(total)[-2])
                    num1 = num1[0:len(num1)-1]
                elif (len(num2) > 0):
                    total = int(num2[-1]) + carry_over
                    answer = str(total)[-1] + answer
                    if total >= 10:
                        carry_over = int(str(total)[-2])
                    num2 = num2[0:len(num2)-1]
        if carry_over != 0:
            print(str(carry_over) + answer)
        else:
            print(answer)                    
    elif (num1[0] == '-' and num2[0] == '-'):
        num1 = num1[1:len(num1)]
        num2 = num2[1:len(num2)]
        
        while (len(num1) > 0 or len(num2) > 0):
            if (len(num1) > 0 and len(num2) > 0):
                total = int(num1[-1]) + int(num2[-1]) + carry_over
                answer = str(total)[-1] + answer
                if total >= 10:
                    carry_over = int(str(total)[-2])
                num1 = num1[0:len(num1)-1]
                num2 = num2[0:len(num2)-1]
            else:
                if (len(num1) > 0):
                    total = int(num1[-1]) + carry_over
                    answer = str(total)[-1] + answer
                    if total >= 10:
                        carry_over = int(str(total)[-2])
                    num1 = num1[0:len(num1)-1]
                elif (len(num2) > 0):
                    total = int(num2[-1]) + carry_over
                    answer = str(total)[-1] + answer
                    if total >= 10:
                        carry_over = int(str(total)[-2])
                    num2 = num2[0:len(num2)-1]        
        if carry_over != 0:
            print('-' + str(carry_over) + answer)
        else:
            print('-' + answer)      
    elif (num1[0] == '-'):
        num1 = num1[1:len(num1)]
        
        while (len(num1) > 0 or len(num2) > 0):
            if (len(num1) > 0 and len(num2) > 0):
                total = int(num2[-1]) - int(num1[-1])
                if total < 0:
                    total += 10
                    num2[-2] = int(num2[-2])-1
                answer = str(total)[-1] + answer

                num1 = num1[0:len(num1)-1]
                num2 = num2[0:len(num2)-1]
            else:
                if (len(num1) > 0):
                    answer = str(total)[-1] + answer
                    num1 = num1[0:len(num1)-1]
                elif (len(num2) > 0):
                    answer = str(total)[-1] + answer
                    num2 = num2[0:len(num2)-1]        
        
        print('-' + answer)   
    elif (num2[0] == '-'):
        num2 = num2[1:len(num2)]
        
        while (len(num1) > 0 or len(num2) > 0):
            if (len(num1) > 0 and len(num2) > 0):
                total = int(num1[-1]) - int(num2[-1])
                if total < 0:
                    total += 10
                    num2[-2] = int(num2[-2])-1
                answer = str(total)[-1] + answer

                num1 = num1[0:len(num1)-1]
                num2 = num2[0:len(num2)-1]
            else:
                if (len(num1) > 0):
                    answer = str(total)[-1] + answer
                    num1 = num1[0:len(num1)-1]
                elif (len(num2) > 0):
                    answer = str(total)[-1] + answer
                    num2 = num2[0:len(num2)-1]        
        
        print('-' + answer)         